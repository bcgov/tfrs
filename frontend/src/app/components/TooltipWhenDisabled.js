import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { OverlayTrigger, Tooltip } from 'react-bootstrap';

class TooltipWhenDisabled extends Component {
  _renderDisabled () {
    return (
      <OverlayTrigger placement="top" overlay={this._tooltip()}>
        <div className="overlay-trigger">
          {this.props.children}
        </div>
      </OverlayTrigger>
    );
  }

  _tooltip () {
    return (
      <Tooltip className="danger" id="tooltip">
        {Array.isArray(this.props.title) &&
          this.props.title.map(title => (<div key={title}>{title}</div>))
        }
        {!Array.isArray(this.props.title) &&
          this.props.title
        }
      </Tooltip>
    );
  }

  render () {
    if (this.props.disabled) {
      return this._renderDisabled();
    }

    return this.props.children;
  }
}

TooltipWhenDisabled.defaultProps = {
  disabled: true
};

TooltipWhenDisabled.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
  disabled: PropTypes.bool,
  title: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string)
  ]).isRequired
};

export default TooltipWhenDisabled;
