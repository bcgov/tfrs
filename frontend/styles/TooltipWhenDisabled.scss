.overlay-trigger {
  display: inline-block;

  button {
    pointer-events: none;
  }
}