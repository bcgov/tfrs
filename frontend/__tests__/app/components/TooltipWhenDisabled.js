import React from 'react';
import renderer from 'react-test-renderer';

import TooltipWhenDisabled from '../../../src/app/components/TooltipWhenDisabled';

test('TooltipWhenDisabled should display properly', () => {
  const component = renderer.create([
    <TooltipWhenDisabled
      disabled
      key="tooltip"
      title="Sample Title"
    >
      <button type="button">Test Button</button>
    </TooltipWhenDisabled>
  ]);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
