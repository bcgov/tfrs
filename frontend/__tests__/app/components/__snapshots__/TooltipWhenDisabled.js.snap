// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipWhenDisabled should display properly 1`] = `
<div
  className="overlay-trigger"
  onBlur={[Function]}
  onClick={null}
  onFocus={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <button
    type="button"
  >
    Test Button
  </button>
</div>
`;
